### PR TITLE
feat: add ability to get and set props on the session object natively

### PR DESCRIPTION
**Pull request by xkal36**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### Overview 
Closes https://github.com/fastify/fastify-secure-session/issues/113

This PR adds the ability to natively set properties on the session object. It does so by proxying the session object.

Note: I proxied the session object in every method i twas returned from, in case these methods are used outside of fastify. I also added this direct getting and setting to every test in order to make sure there were no regressions.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,6 +5,32 @@ const sodium = require('sodium-native')
 const kObj = Symbol('object')
 const kCookieOptions = Symbol('cookie options')
 
+// allows us to directly get and set properties as well as using getters and setters
+const sessionProxyHandler = {
+  get (target, prop) {
+    // Calling functions eg request.session.get('key') or request.session.get('key', 'value')
+    if (typeof target[prop] === 'function') {
```
Comment (mcollina) [inline index.js:12]: nice trick!

**Review by mcollina — APPROVED** [review]
lgtm, good work!

File: README.md [hunk 2]
```diff
@@ -68,6 +68,9 @@ you will see what steps the library is doing and understand why a session you
 expect to be there is not present. For extra details, you can also enable `trace`
 level logging.
 
+Note: instead of using `get` and `set` as seen above, you may also wish to get and set properties on the
+session object natively in order to make your code compatible with other libraries ie `request.session.data = rquest.body` and `const data = request.session.data` are also possible.
```
Comment (jsumners) [inline README.md:72]:
```suggestion
session object natively in order to make your code compatible with other libraries ie `request.session.data = request.body` and `const data = request.session.data` are also possible.
```

File: test/delete.js [hunk 3]
```diff
@@ -14,14 +14,20 @@ fastify.register(require('../'), {
...
 })
 
 fastify.post('/', (request, reply) => {
+  // using setters
   request.session.set('some', request.body.some)
   request.session.set('some2', request.body.some2)
+
+  // setting natively
+  request.session.set('some3', request.body.some3)
+  request.session.set('some4', request.body.some4)
```
Comment (simoneb) [inline test/delete.js:29]: what's the difference between the lines above and these?

Comment (xkal36) [reply test/delete.js:29]: None - good catch - I should be directly setting the props here

**Comment by xkal36** [comment]
Agreed in discussion with @simoneb to remove redundant test code added here and instead test this in isolation

File: README.md [hunk 4]
```diff
@@ -68,6 +68,9 @@ you will see what steps the library is doing and understand why a session you
 expect to be there is not present. For extra details, you can also enable `trace`
 level logging.
 
+Note: instead of using `get` and `set` as seen above, you may also wish to get and set properties on the
+session object natively in order to make your code compatible with other libraries ie `request.session.data = request.body` and `const data = request.session.data` are also possible.
```
Comment (simoneb) [inline README.md:72]: this should be slightly reworded. get and set are methods, so please refer to them as such. the new way you introduced can be described as "using property getters and setters"

File: index.js [hunk 5]
```diff
@@ -5,6 +5,32 @@ const sodium = require('sodium-native')
 const kObj = Symbol('object')
 const kCookieOptions = Symbol('cookie options')
 
+// allows us to directly get and set properties as well as using getters and setters
```
Comment (simoneb) [inline index.js:8]: get and set are methods

**Review by simoneb — APPROVED** [review]
a couple of comments on wording, LGTM otherwise

**Review by simoneb — APPROVED** [review]

**Pull request merged**